src/main/java/org/reactome/addlinks/fileprocessors/MGIFileProcessor.java

```diff
@@ -0,0 +1,52 @@
+package org.reactome.addlinks.fileprocessors;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MGIFileProcessor extends FileProcessor{
+
+    public MGIFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public MGIFileProcessor()
+    {
+        super(null);
+    }
+
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        Path inputFile = Paths.get(this.pathToFile.toAbsolutePath().toString().replace(".gz", ""));
+        try
+        {
+            for (String line : Files.readAllLines(inputFile).stream().collect(Collectors.toList())) {
```

Comment:
It's usually best practice to keep the code included in the `try` block as small as possible by wrapping only the portion that can throw the exception.  In this case, you do write:

```
List<String> lines;
try {
    lines = Files.readAllLines(inputFile);
} catch (IOException e) {
    logger.error("Error reading file ({}): {}", inputFile.toString(), e.getMessage());
    e.printStackTrace();
    return mappings;
}
```
followed by the `for` loop outside of the `try` block.

Reply:
Done with all File Processors now